packages/kit/src/pages.ts

```diff
@@ -34,3 +35,35 @@ export function extendRouteRules (route: string, rule: NitroRouteConfig, options
       : defu(opts.routeRules[route], rule)
   }
 }
+
+export interface AddRouteMiddlewareOptions {
+  /**
+   * Override route rule config
+   *
+   * @default false
+   */
+  override?: boolean
+}
+
+export function addRouteMiddleware (input: NuxtMiddleware | NuxtMiddleware[], options: AddRouteMiddlewareOptions = {}) {
+  const nuxt = useNuxt()
+  nuxt.hook('app:resolve', (app) => {
+    const addMiddleware = (middleware: NuxtMiddleware) => {
+      const find = app.middleware.findIndex(item => item.name === middleware.name)
+      if (find >= 0) {
+        if (options.override === true) {
+          app.middleware[find] = middleware
+        } else {
+          consola.warn(`'${middleware.name}' middleware exists with '${app.middleware[find].path}' path. You can set override option 'true' for replace with your middleware path.`)
```

Comment:
We can use console.warn. It should be wrapped in consola automatically by Nuxt.


```suggestion
          console.warn(`'${middleware.name}' middleware exists with '${app.middleware[find].path}' path. You can set override option 'true' for replace with your middleware path.`)
```